packages/jsii-pacmak/lib/targets/golang/types/interface.ts

```diff
@@ -1,55 +1,102 @@
-import { GoType } from './go-type';
-import { InterfaceType, Method, Property, TypeReference } from 'jsii-reflect';
 import { CodeMaker } from 'codemaker';
+import { InterfaceType, Method, Property } from 'jsii-reflect';
+import { GoType, GoEmitter } from './go-type';
+import { GoTypeRef } from './go-type-reference';
+import { Module } from '../module';
 
-export class TypeMapper {
-  public constructor(public value: TypeReference) {}
+export interface InterfaceField {
```

Comment:
See: https://github.com/aws/jsii/pull/1892#discussion_r470903573

Reply:
I agree that multiple classes makes sense. I was thinking though that we would need a parent class (IE: this one) that does the logic of deciphering if the interface is a struct or not and then instantiating and emitting the correct specific class.

Otherwise we will have to add logic to the the `Module` constructor which feels like it shouldn't be concerned with that.

Either way, I minimized the changes to interface.ts (only replacing TypeMapper with GoTypeRef) to help with this.